### DISCUSSION TITLE
tile source

### DISCUSSION THREAD

**Discussion by abbdef** [mapsforge/vtm#1012] [Q&A] [answered]
Hi, Im using the android example ( ver 0.13) 
Im using a pbf file in the phones download/maps/area-file-latest/filename.pbf
the map polyline and location overlays load but not the map, the android debugger says the tilesource is empty.
In the code I see it's looking for a map file but I used to use pbf files.
I need some clues  , the only thing Ive changed is android 10 compatibility but im building to api 28 and running on and older api26 phone.

**Comment by devemux86** [comment, marked as answer]
Offline maps are supported with the [Mapsforge map format](https://github.com/mapsforge/mapsforge/blob/master/docs/Specification-Binary-Map-File.md).

See the [getting started guide](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/GettingStarted.java).
Mapsforge maps can be downloaded from many [providers](https://github.com/mapsforge/vtm/blob/master/docs/Mapsforge-Maps.md).
Or can [create Mapsforge maps](https://github.com/mapsforge/mapsforge/blob/master/docs/MapCreation.md) with map-writer.

**Comment by abbdef** [comment]
Thanks that works, my mistake was due to the map file and the pbf file coincidentally having the exact same size.
I assume it's safe to continue to use the android download folder > api 30 ?

**Reply by devemux86** [reply]
If you mean the Android scoped storage, see the [Android documentation](https://developer.android.com/about/versions/11/privacy/storage).

VTM can load maps and themes as streams (see [MapsforgeActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MapsforgeActivity.java) example),
if need to read them outside of the app's internal storage.
